src/protocol/DocumentClient.js

```diff
@@ -61,6 +63,25 @@ define('polymer-designer/protocol/DocumentClient', function() {
     /**
      * @returns {Promise}
      */
+    getElementsAtPoint(x, y) {
+      return this.connection.request({
+        messageType: 'getElementsAtPoint',
+        x: x,
+        y: y,
+      }).then(function(response) {
+        let elements = response.elements;
+        for (let i in elements) {
+          let element = elements[i];
+          let id = element.elementInfo.id;
+          this.nodes.set(id, element);
```

Comment:
Interesting. Can you explain what this iteration over `elements` is doing? I have some guesses, but I would love to hear more.